product_template_pricelist_button/models/product_template.py

```diff
@@ -0,0 +1,31 @@
+# © 2024 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (https://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models
+
+class ProductTemplate(models.Model):
+    _inherit = 'product.template'
+
+    def open_pricelist_rules(self):
+        action = super().open_pricelist_rules()
+        self.ensure_one()
+        domain = ['|','|',
+            ('product_tmpl_id', '=', self.id),
+            ('product_id', 'in', self.product_variant_ids.ids),
+            ('categ_id','parent_of',self.categ_id.id),
+            ]
+        action["domain"] = domain
+        return action
+
+    def _compute_item_count(self):
+        super()._compute_item_count()
+        for template in self:
+            template.pricelist_item_count = self.env['product.pricelist.item'].search_count([
+                '&',
+                ('pricelist_id.active', '=', True),
+                '|','|',
+                ('product_tmpl_id', '=', template.id),
+                ('product_id', 'in', template.product_variant_ids.ids),
+                ('categ_id','parent_of',template.categ_id.id),
+            ])
```

Comment:
Also use here `expression.OR`... in this case the original condition is `(A or B) and C` (where C is the active condition). Past and overwrite could be prevented adding the condition the following:
`[(A or B) and C] or [D and C]`
when D is the new `categ_id` condition. C condition should be added as well in this case, due to this code reorganization.